src/components/AddressSearch.js

```diff
@@ -23,134 +23,127 @@ const propTypes = {
-        if (isAddressValidForVBA(addressComponents)) {
+        if (addressComponents) {
             // Gather the values from the Google details
-            const streetNumber = getAddressComponent(addressComponents, 'street_number', 'long_name');
-            const streetName = getAddressComponent(addressComponents, 'route', 'long_name');
-            let city = getAddressComponent(addressComponents, 'locality', 'long_name');
-            if (!city) {
-                city = getAddressComponent(addressComponents, 'sublocality', 'long_name');
-                Log.hmmm('[AddressSearch] Replacing missing locality with sublocality: ', {address: details.formatted_address, sublocality: city});
+            const streetNumber = getAddressComponent(addressComponents, 'street_number', 'long_name') || '';
+            const streetName = getAddressComponent(addressComponents, 'route', 'long_name') || '';
+
+            const addressStreet = `${streetNumber} ${streetName}`.trim();
+
+            let addressCity = getAddressComponent(addressComponents, 'locality', 'long_name');
+            if (!addressCity) {
+                addressCity = getAddressComponent(addressComponents, 'sublocality', 'long_name');
+                Log.hmmm('[AddressSearch] Replacing missing locality with sublocality: ', {address: details.formatted_address, sublocality: addressCity});
             }
-            const state = getAddressComponent(addressComponents, 'administrative_area_level_1', 'short_name');
-            const zipCode = getAddressComponent(addressComponents, 'postal_code', 'long_name');
-
-            // Trigger text change events for each of the individual fields being saved on the server
-            props.onChangeText('addressStreet', `${streetNumber} ${streetName}`);
-            props.onChangeText('addressCity', city);
-            props.onChangeText('addressState', state);
-            props.onChangeText('addressZipCode', zipCode);
-        } else {
-            // Clear the values associated to the address, so our validations catch the problem
-            Log.hmmm('[AddressSearch] Search result failed validation: ', {
-                address: details.formatted_address,
-                address_components: addressComponents,
-                place_id: details.place_id,
-            });
-            props.onChangeText('addressStreet', null);
-            props.onChangeText('addressCity', null);
-            props.onChangeText('addressState', null);
-            props.onChangeText('addressZipCode', null);
+            const addressZipCode = getAddressComponent(addressComponents, 'postal_code', 'long_name');
+            const addressState = getAddressComponent(addressComponents, 'administrative_area_level_1', 'short_name');
+
+            this.props.onChange(_.pick({
+                addressCity,
+                addressState,
+                addressZipCode,
+                addressStreet,
+            }, _.identity));
```

Comment:
We are trying to pass the key and value for all of these properties to `props.onChange`. What's the value of doing it this way?

Reply:
The object passed to `props.onChange` will optionally have the following keys: `addressCity`, `addressState`, `addressZipCode` and `addressStreet`. The `_pick(..., _.identity)` will remove any of they keys that don't have a `truthy` value associated. Why? because if the clicked result is missing some part of the address (i.e. city), we are not going to overwrite what the user may have typed.

Now that you mentioned, the intention is not obvious by looking at the code.

I could add a comment to make it more clear, what do you think?

Reply:
Adding a comment would be very helpful in this case, I missed the overwriting part but that makes sense.

Reply:
NAB, maybe would be better without the underscore `identity` trick and check for each value

```js
const values = {};
if (addressCity) {
    values.addressCity = addressCity;
}

// etc

if (_.size(values) === 0) {
    return;
}

this.props.onChange(values);
```